datastore_mongo.go

```diff
@@ -85,6 +85,23 @@ func (db *DataStoreMongo) GetDevice(id DeviceID) (*Device, error) {
 	return &res, nil
 }
 
+func (db *DataStoreMongo) DeleteDevice(id DeviceID) error {
+	s := db.session.Copy()
+	defer s.Close()
+
+	filter := bson.M{"id": id}
```

Comment:
stacked two commits on top:
- one with the index
- another with a possible travis fix (suggested by GregD) - the build keeps failing, same for devauth @bboozzoo